[AV2-925] Update Forget Password API Call

sasangachathumal opened the pull request:
added new stack objects in unit test.
added forgot password to auth service.

**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-925

**Additional Comments:**
`Add here`

**Checklist:**

- [x] Unit test completed?: (Y)
- [x] Docs folder up to date?: (Y)

All checklists are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

sasangachathumal committed:
[AV2-925] added new stack objects in unit test.
added forgotpassword to auth service.

shawnm0705 commented on src/app/auth/auth.service.ts:
```diff
@@ -227,10 +227,23 @@ export class AuthService {
-    });
+    const body = {
+      email,
+      resetLink: this._createResetDirectLinks(),
+      directLink: this._createResetDirectLinks('direct')
```
why not just separate them into two different functions.
Or make the function name more generic like `_createLinks`

sasangachathumal committed:
[AV2-925] remove s3 paths from stack object.
update naming of login methods in auth service.
modify link create method for forget password.
update storage service.
update getEndpointUrl.

shawnm0705 commented on src/app/shared/request/request.service.ts:
```diff
@@ -96,7 +96,7 @@ export class RequestService {
 
   private getEndpointUrl(endpoint, isLoginAPI?: boolean) {
-    let endpointUrl = this.prefixUrl + endpoint;
+    let endpointUrl = ( this.storage.stackConfig ? this.storage.stackConfig.coreApi : this.prefixUrl ) + endpoint;
```
there is no `this.prefixUrl` anymore, it should be removed from here an the app module as well.
If there is no stack url, we should throw error.

sasangachathumal committed:
[AV2-925] updated the getEndpointUrl method.
